File: utils/sort_by_date_test.go
package utils

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestSortByDate(t *testing.T) {
	input := [][]string{
    {"Date", "Letter", "Number"},
		{"1983-01-07", "a", "1"},
		{"2007-12-19", "b", "2"},
		{"1998-08-27", "c", "3"},
	}

	sorted := SortByDate(input)
	expected := [][]string{
    {"Date", "Letter", "Number"},
		{"1983-01-07", "a", "1"},
		{"1998-08-27", "c", "3"},
		{"2007-12-19", "b", "2"},
	}
	assert.Equal(t, expected, sorted)
}

func TestSortByDate_SortHeaderToTop(t *testing.T) {
	input := [][]string{
		{"1983-01-07", "a", "1"},
		{"2007-12-19", "b", "2"},
		{"1998-08-27", "c", "3"},
    {"Date", "Letter", "Number"},
	}

	sorted := SortByDate(input)
	expected := [][]string{
    {"Date", "Letter", "Number"},
		{"1983-01-07", "a", "1"},
		{"1998-08-27", "c", "3"},
		{"2007-12-19", "b", "2"},
	}
	assert.Equal(t, expected, sorted)
}

File: utils/sortByDate.test.ts
import { expect } from "@std/expect";
import { describe, it } from "@std/testing/bdd";
import { sortByDate } from "./sortByDate.ts";

describe("sortByDate", () => {
  it("sorts dates in ascending order", () => {
    const input = [
      ["Date", "Letter", "Number"],
      ["1983-01-07", "a", "1"],
      ["2007-12-19", "b", "2"],
      ["1998-08-27", "c", "3"],
    ];

    const sorted = sortByDate(input);
    const expected = [
      ["Date", "Letter", "Number"],
      ["1983-01-07", "a", "1"],
      ["1998-08-27", "c", "3"],
      ["2007-12-19", "b", "2"],
    ];

    expect(sorted).toEqual(expected);
  });

  it("sorts header to top", () => {
    const input = [
      ["1983-01-07", "a", "1"],
      ["2007-12-19", "b", "2"],
      ["1998-08-27", "c", "3"],
      ["Date", "Letter", "Number"],
    ];

    const sorted = sortByDate(input);
    const expected = [
      ["Date", "Letter", "Number"],
      ["1983-01-07", "a", "1"],
      ["1998-08-27", "c", "3"],
      ["2007-12-19", "b", "2"],
    ];

    expect(sorted).toEqual(expected);
  });
});
